pyomo/contrib/mpc/interfaces/model_interface.py

```diff
@@ -397,6 +416,53 @@ def get_tracking_cost_from_constant_setpoint(
             variables, time, setpoint_data, weight_data=weight_data
         )
 
+    def get_tracking_cost_from_target_trajectory(
+        self,
+        target_data,
+        time=None,
+        variables=None,
+        weight_data=None,
+        context=None,
+    ):
+        if time is None:
+            time = self.time
+        if isinstance(target_data, (Var, Param, Expression)):
+            if variables is None:
+                raise RuntimeError(
+                    "Variables must be provided if we are using a Param or"
+                    " Expression as an argument"
+                )
+            target_data = TimeSeriesData.from_pyomo_components(
+                variables,
+                target_data,
+                time,
+                time_set=self.time,
+                context=context,
+            )
+        elif not isinstance(target_data, TimeSeriesData):
+            target_data = TimeSeriesData(*target_data, time_set=self.time)
```

Comment:
Sometimes I'm really amazed how you can think of something super creative......
It's nice here to use `TimeSeriesData` to save measurement components.